pkg/workspace/library.go

```diff
@@ -15,6 +15,7 @@ const (
 )
 
 type Library struct {
+	parent   *Library
```

Comment:
let's also kick Library changes into other PRs, as they are not necessary for this refactor.

Reply:
I need it to construct the relative path.

The relative path in file.go is not enough, because the relative path at file loading time is not the same as the relative path at evaluation time in case the evaluated library is not the root library.

I can probably avoid this by constructing the relative path at evaluation time though. I'll try that.